zebra/if_netlink.c

```diff
@@ -1302,11 +1302,29 @@ int netlink_interface_addr(struct nlmsghdr *h, ns_id_t ns_id, int startup)
 
 	ifp = if_lookup_by_index_per_ns(zns, ifa->ifa_index);
 	if (ifp == NULL) {
-		flog_err(
-			EC_LIB_INTERFACE,
-			"netlink_interface_addr can't find interface by index %d",
-			ifa->ifa_index);
-		return -1;
+		if (startup) {
+			/* During startup, failure to lookup the referenced
+			 * interface should not be an error, so we have
+			 * downgraded this condition to warning, and we permit
+			 * the startup interface state retrieval to continue.
+			 * Prior to downgrading this to a warning, returning an
```

Comment:
I know this is a minor nit, but anyone reading this code won't really have the context to understand what "Prior to" references without checking the git history anyway. I would remove the "Prior to downgrading this to a warning, " part. The rest is good.